### ISSUE TITLE
cyclic-import warning detects a non-issue situation

### ISSUE THREAD

**Issue by praiskup** [PyCQA/pylint#6983]
### Bug description

```
shell>$ ls pylint_mod/
a.py  b.py  __init__.py

shell>$ cat pylint_mod/a.py 
import pylint_mod.b

def x():
    print("In A module")

def test():
    pylint_mod.b.x()

shell>$ cat pylint_mod/b.py 
import pylint_mod.a

def x():
    print("In B module")

def test():
    pylint_mod.a.x()
```

The problem is that this warning is currently printed by pylint:

```
$ pylint pylint_mod/a.py pylint_mod/b.py  | grep -i cycl
pylint_mod/b.py:1:0: R0401: Cyclic import (pylint_mod.a -> pylint_mod.b) (cyclic-import)
```

While this technically _is_ a cyclic-import, practically it should be a non-issue.  Both the cyclic cross-module uses are delayed to the run-time, therefore all the module objects are correctly loaded when really needed.

### Command used

```shell
pylint pylint_mod/a.py pylint_mod/b.py
```

### Pylint output

```shell
pylint_mod/b.py:1:0: R0401: Cyclic import (pylint_mod.a -> pylint_mod.b) (cyclic-import)
```

### Expected behavior

No error printed.

### Pylint version

```shell
pylint 2.13.7
astroid 2.11.5
```

**Comment by Pierre-Sassoulas** [comment]
``cyclic-import`` is a refactor message, not an error message it's acting as intended in this case. (i.e. it's a style check, it's not supposed to warn you only of something that will crash). It's possible to disable the check if you don't like it.

**Comment by praiskup** [comment]
The thing is that we have to turn this warning ON or OFF, only because of this issue.  While typically it provides useful info.

**Comment by Pierre-Sassoulas** [comment]
See https://github.com/PyCQA/pylint/issues/850

**Comment by praiskup** [comment]
> You need to be able to disable it locally though (in a function or elsewhere) without disabling for the whole project.

To be able to do this, we'd have to switch to a function-local import, which is something we don't want to.
Therefore we **have to** disable this for the whole project.

**Comment by DanielNoord** [comment]
@praiskup What @Pierre-Sassoulas is talking about is that we should fix `cyclic-import` to be disable-able per line/module/scope.

It's in the spirit of the message to report on cases such as these, and as explained in the linked message it doesn't make a lot of sense to change the spirit of the message.
That said, the message is not as controllable as we would like and how we allow control for other messages. That's something that should be worked on.

After that has been merged you should be able to add a disable to module/lines where you want to use this pattern.
